src/backend/metal/src/device.rs

```diff
@@ -254,7 +267,7 @@ impl Device {
         debug!("SPIRV-Cross generated shader:\n{}", shader_code);
 
         let options = metal::CompileOptions::new();
-        options.set_language_version(MTLLanguageVersion::V1_1);
+        options.set_language_version(MTLLanguageVersion::V1_2);
```

Comment:
Does this affect support for older devices? Do we need a fallback?

Reply:
I'm not sure, I've had trouble locating a comprehensive compatibility table for the Metal language version.

Eventually we'll probably have to request the highest version we can and use supported capability flags, like the OpenGL backend had. We could only request `V1_2` when a specialization constant value is provided, but I don't know if it's worthwhile at the moment.

Reply:
https://developer.apple.com/metal/Metal-Feature-Set-Tables.pdf is the table I've been looking at so far, but it lists OS versions only. I don't know much about device compatibility with the various iOS/tvOS/macOS versions, or whether we have a minimum target for OS version.

Reply:
It seems to require a newer OS version at least:
https://developer.apple.com/documentation/metal/mtllanguageversion/1515361-version1_1
https://developer.apple.com/documentation/metal/mtllanguageversion/1639961-version1_2